core/src/main/java/io/lionweb/lioncore/java/serialization/PrimitiveValuesSerialization.java

```diff
@@ -3,15 +3,26 @@
 import com.google.gson.Gson;
 import com.google.gson.JsonElement;
 import com.google.gson.JsonParser;
+import io.lionweb.lioncore.java.metamodel.Enumeration;
+import io.lionweb.lioncore.java.metamodel.EnumerationLiteral;
 import io.lionweb.lioncore.java.metamodel.LionCoreBuiltins;
-import java.util.HashMap;
-import java.util.Map;
+import io.lionweb.lioncore.java.metamodel.Metamodel;
+import java.util.*;
 
 /**
  * This class is responsible for serialization and unserializing primitive values, based on the type
  * of the primitive value.
  */
 public class PrimitiveValuesSerialization {
+
+  private Map<String, Enumeration> enumerationsByID = new HashMap<>();
+
+  public void registerMetamodel(Metamodel metamodel) {
+    metamodel.getElements().stream()
+        .filter(e -> e instanceof Enumeration)
+        .forEach(e -> enumerationsByID.put(e.getID(), (Enumeration) e));
```

Comment:
enumeration keys are only required to be unique within the language, so we could have a key clash here.

Reply:
Yes, but here in enumerationsByID we want to use the ID and not the key, so that we are able to retrieve the enumeration definitions easily. We then use the key for the value. EnumerationsByID is only an internal structure used by PrimitiveValuesSerialization (I am adding a comment about that)

Reply:
To clarify: yes, we use the enumeration key when figuring out which enumeration we are dealing with. So at some point we get an Enumeration instance (i.e., we have access to the name, id, key, etc.).

Then in `serializePropertyValue` we call the `primitiveValuesSerialization` in this way:
```
primitiveValuesSerialization.serialize(dataType.getID(), value);
```

There we pass the primitive type ID, because it is unique. It is not the value we use to figure out the primitive type, but just as unique identifier to register and then retrieve the primitive serializers and unserializers

Reply:
You're right.